### ISSUE TITLE
Problem with Mars.2 not showing workspace label

### ISSUE THREAD

**Issue by KarlEilebrecht** [turesheim/eclipse-launcher#5279]
After using your awesome plugin for a long time with Eclipse Mars.1, I upgraded (Eclipse-internal-update) to Mars.2. Since this time the tiny workspace label does no longer stay(!) on top of the eclipse logo on the Mac OS dock. In other words: after showing up for a moment it vanishes. 
I uninstalled the eclipse-launcher and re-installed it - no change.
Is there any workaround?

Confirmed: Same problem for my colleage right after updating to Mars.2.

**Comment by turesheim** [comment]
How did you install Eclipse? I've seen that Oomph sometimes creates problems with the label.

**Comment by KarlEilebrecht** [comment]
I downloaded Eclipse Mars.1 edition for Java development from eclipse.org<http://eclipse.org>. Later I updated to Mars.2 inside Eclipse. Same for my colleague.

Am 11.03.2016 um 20:32 schrieb Torkild U. Resheim <notifications@github.com<mailto:notifications@github.com>>:


How did you install Eclipse? I've seen that Oomph sometimes creates problems with the label.

-

**Comment by turesheim** [comment]
Could you please check the log for any exceptions that are suspicious? Also make sure that you're running Eclipse under Java 1.7 or newer.

**Comment by KarlEilebrecht** [comment]
eclipse.buildId=4.5.2.M20160212-1500
Java version is 1.8.0_72 (Oracle)

workspace/.metadata/.log does not contain any related error or warning.

**Comment by turesheim** [comment]
I attempted to reproduce the issue by the following procedure:

1. Install "Eclipse IDE for Java Developers" Mars.1
2. Install "OS X Eclipse Launcher Utility"
3. Set the workspace name to "TEST"
4. Upgrade to "Eclipse IDE for Java Developers" Mars.2

I was not able to reproduce the problem. Could there be some special characters in the workspace name that causes this? Either in the file path or in the preference. I tried some Unicode for fun but could not break it. If you send me your .log file I might be able to spot the problem.
<img width="85" alt="i-heart-eclipse" src="https://cloud.githubusercontent.com/assets/337454/13731208/dfe98ffe-e963-11e5-946b-6f6e0e686919.png">

**Comment by turesheim** [comment]
For the record, what OS X version are you using.

**Comment by KarlEilebrecht** [comment]
Hi,

Thank you for your quick reply.

Just to be sure not to be on the wrong page, I just took my other mac, already on El Capitain and completely unrelated to my usual work and used "Eclipse -> Help -> Check for Updates", ran the update, restarted and got the exact same behavior: the label above the ecliipse Icon appears at start and vanishes a moment (maybe 1 or 2 seconds) later. 

My Mac at work, where I observed the problem first, is Yosemite 10.10.5

At  home (same problem): El Capitan 10.11.3

Here is the .log-file from the computer (El Capitan) I just upgraded:
https://www.evernote.com/l/AIz4j2fdZdNMcrjrXoI_2ykyPrFrMDaVotk

**Comment by turesheim** [comment]
The log file contained nothing interesting. Could you try opening a new fresh workspace and see what happens? What other plug-ins do you have installed?

**Comment by turesheim** [comment]
I've improved the way the WorkspaceDecorator obtains the handle to the TaskBar and TaskItem. Please try with the plug-ins installed at https://resheim.net/p2/eclipse-launcher_test/.

**Comment by KarlEilebrecht** [comment]
Unfortunately the new version shows same behavior. After a few seconds the label is gone.
Here I have attached now my full configuration:
https://www.evernote.com/l/AIz4j2fdZdNMcrjrXoI_2ykyPrFrMDaVotk

**Comment by FoxyBOA** [comment]
Have the same problem. Icon (badge) appears sometimes (mainly after -clean option) but most of time it's invisible.

**Comment by turesheim** [comment]
I'm looking into this. Still trying to reproduce the behavior. Thanks for the configuration Karl. I did not see anything strange in there except that Oomph is installed. I'll have a close look at the Oomph source code as I suspect that it does attempts to set the icon badge when on OS X.

**Comment by turesheim** [comment]
Yes, so Oomph appears to interfering in it's ```TaskItemDecorator```. However it only uses the workspace name preference setting. So if the name has not been set, the badge will be cleared. Try actively setting the workspace name in *General > Workspace > Workspace name*.

**Comment by bee256** [comment]
Thanks, Torkild, actively setting the workspace name did the trick. Field was empty…
Cheers, Bernd

**Comment by KarlEilebrecht** [comment]
Fantastic! Made my day! :-)

**Comment by richardliechty** [comment]
Thanks for the tip. Setting the workspace name is in someways cleaner than long workspace folder names that get trun…ted.

**Comment by turesheim** [comment]
I think I can close this as resolved although Oomph will interfere.

**Comment by F4154LK** [comment]
I've just done the upgrade and have the same issue. Manually inserting the names of workspaces is a good work around but not the solution. Will I always have to manually put this in all the time now?

**Comment by turesheim** [comment]
Yes, until I can convince the Oomph team to not override what this plug-in does or implement the same mechanism for showing the badge. Sorry about that.

**Comment by colemarkham** [comment]
Do you have a bug on Oomph? I'd be happy to comment/vote there.

**Comment by turesheim** [comment]
I've added bug https://bugs.eclipse.org/bugs/show_bug.cgi?id=496450

**Comment by jfrantzius** [comment]
It seems that the workaround doesn't work around anymore with Neon.2, at least for me. Does it still work for somebody else in Neon.2?

**Comment by turesheim** [comment]
Yes, it's broken. Probably for other reasons. See bug #5281.

**Comment by takacsot** [comment]
My temporary workaround:
setting: General > Startup and Shutdown
And there I have unchecked Oomph related pugins (not to activate on startup.

**Comment by edfink** [comment]
FYI, the badge wasn't showing for me on Oxygen.3a. Disabling Oomph as @takacsot mentioned above "fixed" it for me.
